lib/bandit/http1/handler.ex

```diff
@@ -63,8 +63,15 @@ defmodule Bandit.HTTP1.Handler do
   end
 
   defp build_transport_info(socket) do
-    {ThousandIsland.Socket.secure?(socket), ThousandIsland.Socket.local_info(socket),
-     ThousandIsland.Socket.peer_info(socket), ThousandIsland.Socket.telemetry_span(socket)}
+    secure? = ThousandIsland.Socket.secure?(socket)
+    telemetry_span = ThousandIsland.Socket.telemetry_span(socket)
+
+    with local_info when is_map(local_info) <- ThousandIsland.Socket.local_info(socket),
```

Comment:
Should we keep local_info/1 and peer_info/1 return signatures closer to their counterparts from OTP library? Is so, I could do that later this week.

Reply:
In terms of the map structures, you mean? 

I'd originally tried to cut them in the shape of the `Plug.Conn.Adapter.get_peer_data` call (and had managed to push those same semantics all the way down to the socket calls), but now that that's been forgone by the case block in this PR, I don't have any problem going the other way and pulling the signatures up from OTP. 

Or did you mean something else entirely? :)

Reply:
That's exactly what I was wondering about :smiley:

Reply:
Great. Let's get this landed and we can work that up as a separate PR. 

To that end, care to review what's here?